docs/proposals/registry/registry-operator.md

```diff
@@ -0,0 +1,97 @@
+
+## Problem
+Currently, the devfile registry is deployed onto a running cluster by `kubectl apply`-ing a few yaml files. This works fine for a proof of concept or doing a quick demo, but there’s a few problems with this approach that make it infeasible long-term:
+
+1) Configuration is difficult.
+   
+   For example, the current approach makes it difficult to programatically configure TLS and authentication for the OCI registry server when deploying.
+
+2) Deployment isn't user friendly and hard to tell how/why something went wrong
+
+3) Updates to the registry instance are not straightfoward
+
+    Requires the registry administrator to manually edit and update the devfile registry deployment resource.
+
+3) Deployment isn’t the same on Kubernetes and OpenShift.
+
+   To handle this, we currently have to maintain two separate sets of yaml files for deploying the registry on Kubernetes or OpenShift.
+
+## Solution
+Implementing an operator to programatically deploy devfile registries will streamline the deployment process on both Kubernetes and OpenShift, as well as provide greater flexibility for customization and configuration when deploying it (especially as we add support for deploying with TLS and authentication enabled). It will also provide us the ability to onboard the devfile registry to OperatorHub in the future, if we wish.
+
+## DevfileRegistry Custom Resource
+
+### API Group:
+`registry.devfile.io`
+
+### API Version
+`v1alpha1`
+
+### Resource Kind
+`DevfileRegistry`
+
+### Scope
+`Namespaced`
+
+### Fields
+
+#### Registry 
+`devfileIndexImage` (string) Image used to bootstrap the OCI registry with devfiles and host the index.json.
+
+`ociRegistryImage` (string) Image used for the OCI registry instance. Defaults to `registry:2.7.1`
+
+#### Storage
+`enabled` (bool) Enables persistent storage for the registry. Defaults to true.
+
+`registryVolumeSize` (string). Size of the volume to use. Defaults to 3Gi.
+
+#### TLS
+`enabled` (bool) TLS mode for the devfile registry. Defaults to true.
+
+`tlsSecretName` (string) Kubernetes secret used for TLS termination on the ingress resource
+
+#### Kubernetes Specific
+`ingressDomain` (string) Ingress domain to be used when deploying on Kubernetes.
+
+### DevfileRegistry Custom Resource Formatting
+```
+apiVersion: registry.devfile.io/v1alpha1
+kind: DevfileRegistry
+metadata:
+  name: registry
+spec:
+  bootstrapImage: ''
+  ociRegistryImage: ''
+  tls:
+    enabled: true
+    secretName: ''
+  storage:
+    enabled: true
+    registryVolumeSize: '3Gi'
+  k8s:
+    ingressDomain: ''
+```
+
+## Operator Functionality
+
+### `DevfileRegistry` Creation
+When an instance of a `DevfileRegistry` resource is created in the namespace where the Devfile Registry operator is deployed, the following occurs in the operator's `reconcile` loop:
+
+1) A service exposing the devfile index server and OCI registry is created
+2) If persistence was enabled, The service that exposes the registry is created
```

Comment:
Should this make reference to a PVC instead?

Reply:
D'oh. I did some reformatting of the checklist and messed things up haha. I'll go fix that.